### PR TITLE
Export two new variables: DESTDIR and DB_DUMP_DEBUG

### PR DESCRIPTION
**Pull request by juanluisbaptiste**
Added two new variables:

- DESTDIR: To export the destination directory so for example you can copy a new tarball including some other files along with the sql dump.

- DB_DUMP_DEBUG: To enable debug mode in post-backup scripts.

### CODE REVIEW

**Comment by juanluisbaptiste** [comment]
Also the post-backup script was wrong according to the exported variables, so I updated that too.

**Comment by deitch** [comment]
Thanks for this @juanluisbaptiste . These are nice additions.

Got a few questions:

1. `DB_DUMP_DEBUG` is basically to enable debug in `post-backup` scripts? Just a flag that a script can use? I think that the name would make more sense as `POST_BACKUP_DEBUG`, and then just be passed to the script as `DEBUG`. So something like:

```sh
NOW=${now} DUMPFILE=${TMPDIR}/${TARGET} DEBUG=${POST_BACKUP_DEBUG} $i
```

and then in a post-backup script:

```sh
if [[ -n "$DEBUG" ]]; then
  set -x
fi
```

Keeps it clean and simple for those scripts, but has a specific var for passing the value to the container itself.

2. `DESTDIR`: I don't get it. What does this add? Mind sharing an actual example?
3. Fixing the backup example: nice catch, thank you.

**Comment by deitch** [comment]
Sorry, hit "Close" instead of "Comment". Reopened.

File: scripts.d/post-backup/rename_backup.sh.example [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 then
-  mv ${TMPDIR}/${TARGET} ${TMPDIR}/${new_name}
-  TARGET=${new_name}
+  now=$(date +"%Y-%m-%d-%H_%M")
+  new_name=db_backup-${now}.gz
+  old_name=$(basename ${DUMPFILE})
+  echo "Renaming backup file from ${old_name} to ${new_name}"
+  mv ${DUMPFILE} ${DESTDIR}/${new_name}
 else
   echo "ERROR: Backup file ${TMPDIR}/${TARGET} does not exist!"
```
Comment (deitch) [inline scripts.d/post-backup/rename_backup.sh.example]: Shouldn't this also be:

```sh
echo "ERROR: Backup file ${DUMPFILE} does not exist!"
```

**Comment by juanluisbaptiste** [comment]
About your suggestion of the debug env variable name, changing it to a different one wouldn't mean having to use two variables instead of just one ?

About DESTDIR, think of for example creating a new tarball containing the sql dump gz and another tarball with a website' s content. If you don't have DESTDIR, where do you put the new tarball ?

Example that is coming in another PR:

https://github.com/juanluisbaptiste/mysql-backup/blob/wordpress_backup_script/scripts.d/post-backup/backup_wordpress_root.sh.example

File: scripts.d/post-backup/rename_backup.sh.example (hunk 1 above)
Comment (juanluisbaptiste) [reply scripts.d/post-backup/rename_backup.sh.example]: oops missed that one.

Comment (juanluisbaptiste) [reply scripts.d/post-backup/rename_backup.sh.example]: Done.

**Comment by deitch** [comment]
> About your suggestion of the debug env variable name, changing it to a different one wouldn't mean having to use two variables instead of just one ?

Oh, I see. You aren't creating a new var, but instead reusing the existing one `DB_DUMP_DEBUG`, but are adding that it gets passed to the post-backup scripts. OK, got it. Let's leave it how you did it. Makes sense once I grasped it.

**Comment by deitch** [comment]
> About DESTDIR, think of for example creating a new tarball containing the sql dump gz and another tarball with a website' s content. If you don't have DESTDIR, where do you put the new tarball ?

Don't you already have `DB_DUMP_TARGET`? The README example shows:

```
docker run -d --restart=always -e DB_USER=user123 -e DB_PASS=pass123 -e DB_DUMP_FREQ=60 -e DB_DUMP_BEGIN=2330 -e DB_DUMP_TARGET=/db --link my-db-container:db -v /local/file/path:/db deitch/mysql-backup
```

So it will dump it into `/db` (as seen from inside the container), which is whatever you mapped it to from outside the container, in the above example `/local/file/path`. Could you not just then do your alternate website copy there? E.g.

```
$ docker run -d --restart=always -e DB_USER=user123 -e DB_PASS=pass123 -e DB_DUMP_FREQ=60 -e DB_DUMP_BEGIN=2330 -e DB_DUMP_TARGET=/db --link my-db-container:db -v /local/file/path:/db deitch/mysql-backup
$ tar czvf /local/file/path/web_backup.tgz /var/html
```

**Comment by juanluisbaptiste** [comment]
So it will dump it into /db (as seen from inside the container), which is whatever you mapped it to from outside the container, in the above example /local/file/path. Could you not just then do your alternate website copy there? E.g.

> $ docker run -d --restart=always -e DB_USER=user123 -e DB_PASS=pass123 -e DB_DUMP_FREQ=60 -e DB_DUMP_BEGIN=2330 -e DB_DUMP_TARGET=/db --link my-db-container:db -v /local/file/path:/db deitch/mysql-backup
> $ tar czvf /local/file/path/web_backup.tgz /var/html

No because what you suggests means doing the website files backup outside the container, so I would not be able to have one tarball with both files and sql dump, created automatically at the same time. 

About DB_DUMP_TARGET, its the same, as I am creating a new tarball in a temp dir, I need to move it to DB_DUMP_TARGET but if I don't pass that value to the script I can't move the new tarball.

**Comment by deitch** [comment]
> No because what you suggests means doing the website files backup outside the container, so I would not be able to have one tarball with both files and sql dump, created automatically at the same time.

You are trying to get the `mysql-backup` image to be able to do backup of a mysql DB _and_ some arbitrary other "something"? 

Assuming this is desired, II still don't get how what you are proposing would make that possible. `mysql-backup` still does a DB dump to a tmp directory, names the file with an appropriate timestamp, zips it up, and places it somewhere else (S3, CIFS, directory, etc.). How does this change that?

**Comment by deitch** [comment]
Maybe a practical example of how it would work?

**Comment by juanluisbaptiste** [comment]
A backup of just the db is not very useful without the application files, for example, a wordpress installation. Why split a backup process in two if it can be done in just one ? the example in question:

https://github.com/juanluisbaptiste/mysql-backup/blob/wordpress_backup_script/scripts.d/post-backup/backup_wordpress_root.sh.example

**Comment by deitch** [comment]
> Why split a backup process in two if it can be done in just one ? the example in question

Ah, so you use the `post-backup/` to do additional backup? I get it. I thought you were going down the path of making the `mysql-backup` itself able to do all of that, which would grow quickly beyond control. Using the pluggable `post-backup` makes a lot of sense.

So `DESTDIR` is just for the `post-backup/` to know where it should dump its data, a convenience var to pass the target to the script?

Now I get it. Makes a lot of sense.

**Comment by deitch** [comment]
Go ahead and bump the minor version number, please, and we can merge it in.

**Comment by juanluisbaptiste** [comment]
> Ah, so you use the post-backup/ to do additional backup? I get it. I thought you were going down the path of making the mysql-backup itself able to do all of that, which would grow quickly beyond control. Using the pluggable post-backup makes a lot of sense.

>So DESTDIR is just for the post-backup/ to know where it should dump its data, a convenience var to pass the target to the script?

>Now I get it. Makes a lot of sense.

Yes, exactly, I'm sorry if I wasn't clear before, I thought I was. This has a lot of potential, we could write a generic version of the wordpress script I made that can backup any sql+files application. I have a bunch of backup script from work for a lot of services that I will migrate to this new plugin model and add to the examples directory soon.

**Comment by juanluisbaptiste** [comment]
>Go ahead and bump the minor version number, please, and we can merge it in.

I'm also working on the pre restore stuff, so a full backup done with a post-backup script can also be restored using this image. This means add pre-restore directory to be able to uncompress the full backup tarball so the sql dump and files can be restored, and  maybe a post-restore directory so files can be restored after sql dump restore if desired.

So bump now with this and bump again later this week when I have the rest ? or wait to have everything done ?

**Comment by juanluisbaptiste** [comment]
Also, what about using DUMPDIR instead of DESTDIR to be more concise with DUMPFILE ?

**Comment by deitch** [comment]
> I'm sorry if I wasn't clear before

No apologies. Just as likely my missing it. :-)

> So bump now with this and bump again later this week when I have the rest ? or wait to have everything done ?

Yeah, bump now. I love small limited-risk changes.

> Also, what about using DUMPDIR instead of DESTDIR to be more concise with DUMPFILE ?

Sure. I like that.

Thanks again.

**Comment by deitch** [comment]
Also needs a rebase, since I (finally) got it into Travis.

**Comment by juanluisbaptiste** [comment]
The bump you mean on the version tag ?

**Comment by deitch** [comment]
> The bump you mean on the version tag ?

I did, but got confused on 2 projects. This one uses hashes, so never mind.

Let's just rebase, switch to DUMPDIR, and merge in.

**Comment by juanluisbaptiste** [comment]
I think it's ready.

**Comment by deitch** [comment]
And CI passes, so away we go!

**Comment by juanluisbaptiste** [comment]
Excellent!

**Pull request merged**